### PR TITLE
[Ready] Makes kitchen cold rooms actually cold (Metastation)

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
tweak: Metastation's kitchen cold room is now really cold! Make sure to wear warm clothing.
tweak: The kitchen clothing vendor now stocks two winter jackets by default.
tweak: Goats are now fine with temperatures as low as 180°K. This is so that Pete doesn't freeze to death inside the cold room.
code: Added a kitchen area subtype for cold rooms and adjusted chef CQC to work inside it (no gameplay changes).
/:cl:

You know how it's called a cold room, looks like a cold room but is actually just another storage room?

I made the Meta kitchen cold room spawn at -80°C and added a freezer to it. Originally I wanted to pick a more realistic temperature (like -25°C), but the way body temperature works means that you'd have to sit inside it for minutes to feel an effect.

Now, you can store bodies inside it, trap people and let them freeze to death, or lower the temperature even further and use it as a ghetto cryo chamber.

Also:
Kitchen clothing vendor gets two winter jackets and goats survive at lower temperatures (so Pete doesn't just die).

yeap

### CODE REVIEW

File: _maps/map_files/MetaStation/MetaStation.dmm [hunk 1]
```diff
@@ -49951,69 +49912,68 @@
...
-	dir = 1
-	},
 /obj/structure/cable/yellow{
 	icon_state = "4-8"
 	},
-/obj/effect/landmark/blobstart,
 /obj/effect/turf_decal/stripes/line{
 	dir = 8
 	},
+/obj/machinery/atmospherics/components/binary/passive_gate{
```
Comment (81Denton) [inline _maps/map_files/MetaStation/MetaStation.dmm]: This one keeps the cold room freezer from cooling the whole distro loop

**Comment by 81Denton** [comment]
Forgot to mention: If this gets merged, I'll change the cold rooms of other maps as well.

**Comment by coiax** [comment]
I don't know if decals have their own internal temperature, but if not, I'll have gibs not generate miasma in low temperatures in #41782.

Then provided the chef's room is refrigerated, it won't rot.

**Comment by yoyobatty** [comment]
I wonder if we could put a fridge next to a pump that is outputting to the air vent in there or a space cooler.

**Comment by 81Denton** [comment]
@Tlaltecuhtli had the good idea of adding H/E pipes instead of a single vent, I'll play around with it when I have time to work on it

**Comment by Steelpoint** [comment]
I swear someone did this before in the distant past. Good idea nonetheless.

**Comment by MACIEKBAKI** [comment]
Is it at such a low temperature to deal damage to someone without proper (winter) clothing?

**Comment by Therapist1** [comment]
This wont cause immense spacewind when I open the door though, will it?

**Comment by kingoftheautism** [comment]
@Therapist1 It shouldn't, since pressure is relatively same, only difference is temperature. Time to scavenge maints to get some space heaters

File: _maps/map_files/MetaStation/MetaStation.dmm (hunk 1 above)
Comment (kingoftheautism) [reply _maps/map_files/MetaStation/MetaStation.dmm]: Or, you know, you could simply place pump like everywhere else. It's not like I'm against that passive gate tho

**Comment by Grazyn** [comment]
Clothing vendor is inside the cold room so you're forced to suffer a bit of freeze damage before you can dispense a coat

**Comment by 81Denton** [comment]
@MACIEKBAKI Yeah, it will start hurting you after a few seconds. That's why I added winter jackets to the chef clothing vendor, so chefs can hang out in their workplace without freezing to death.

@Therapist1 I adjusted the cold room's mole count so it's also about 101.3 kPa, but at a much lower temperature. It shouldn't cause much space wind.

**Comment by Anonmare** [comment]
Slightly worried this might cause some annoying temperature leak every time the door gets opened

**Comment by PKPenguin321** [comment]
How quickly do the affects of the temperature set in? It shouldn't be instant

**Comment by 81Denton** [comment]
I'm gonna replace the vent with H/E piping and do some testing. It might even be better to start it at warmer temperatures so it's not as dangerous as let's say telecomms.

**Comment by Therapist1** [comment]
Perhaps you could just install some tiny fans at the doors as to not let the air leak out into maint or the kitchen

**Comment by coiax** [comment]
Tiny fans barely exist canonically, only to compensate for our lack of complex airlock interlocking air pumps

**Comment by 81Denton** [comment]
Turns out that H/E pipes work really well for regulating room temperatures. You can even replace the freezer with a heater and have yourself a nice BBQ.
I just need to remodel the layout a bit so it looks less ugly.

capture

**Comment by 81Denton** [comment]
Alright, the piping is now black so it clashes less with the H/E color scheme.
I also tested the damage this room does - about 15 damage if you hang around without warm clothing for a minute. This of course changes if you crank up the temperature.

File: code/datums/martial/cqc.dm [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 /datum/martial_art/cqc/can_use(mob/living/carbon/human/H)
 	var/area/A = get_area(H)
-	if(just_a_cook && !(istype(A, /area/crew_quarters/kitchen)))
+	if(just_a_cook && (!(istype(A, /area/crew_quarters/kitchen)) || !(istype(A, /area/crew_quarters/kitchen/coldroom))))
```
Comment (ShizCalev) [inline code/datums/martial/cqc.dm]: if it's not a type of `/area/crew_quarters/kitchen`, or a type of `/area/crew_quarters/kitchen/coldroom`?

Comment (81Denton) [reply code/datums/martial/cqc.dm]: Ah, didn't know that istype includes subtypes as well.

**Comment by optimumtact** [comment]
You've answered nobody's concerns about temperature leakage into other rooms.

Go and fucking find out already.

**Comment by 81Denton** [comment]
@optimumtact From testing it looks like temperature leakage is no issue. I bolted the cold room airlock open for a minute in a row - temp of the surrounding kitchen tiles dropped to temps between ~2.5°C and 10°C. This was not low enough to cause any damage to nearby humans.

During rounds, the airlock won't be open for that long in a row since chefs don't want Pete to escape.

**Comment by Bawhoppen** [comment]
Pete saves the day yet again.

**Pull request merged**